packages/server/api/src/app/flows/issues/issues-service.ts

```diff
@@ -57,20 +77,71 @@ export const issuesService = (log: FastifyBaseLogger) => ({
         })
 
-        const query = repo().createQueryBuilder(IssueEntity.options.name).where({
-            projectId,
-            status: IssueStatus.ONGOING,
-        })
+        let query = repo().createQueryBuilder(IssueEntity.options.name)
+            .where({ projectId })
+
+        if (status && status.length > 0) {
+            query = query.andWhere('status IN (:...statuses)', { statuses: status })
+        }
 
         const { data, cursor: newCursor } = await paginator.paginate(query)
 
-        const populatedIssues = await Promise.all(data.map(async issue => {
+        const issuesWithCounts = await Promise.all(data.map(async issue => {
+            // Fixed query with proper join between issue and flow_run
+            const count = await repo()
+                .createQueryBuilder('i') 
+                .select('COUNT(*)', 'count')
+                .innerJoin('flow_run', 'fr', 'i."flowId" = fr."flowId" AND i."stepName" = fr."failedStepName"')
+                .where('fr."flowId" = :flowId', { flowId: issue.flowId })
+                .andWhere('fr.status IN (:...statuses)', { statuses: [FlowRunStatus.FAILED, FlowRunStatus.INTERNAL_ERROR, FlowRunStatus.TIMEOUT] })
+                .andWhere('fr."failedStepName" = :stepName', { stepName: issue.stepName })
+                .andWhere('i.id = :issueId', { issueId: issue.id })
+                .getRawOne()
+                .then(result => parseInt(result.count, 10))
+
+            if (count === 0 && issue.status === IssueStatus.UNRESOLVED) {
+                await this.updateById({
+                    projectId: issue.projectId,
+                    id: issue.id,
+                    status: IssueStatus.RESOLVED,
+                })
+                issue.status = IssueStatus.RESOLVED
+            }
+
+            const actualStatus = issue.status === IssueStatus.ARCHIVED 
+                ? IssueStatus.ARCHIVED 
+                : (count === 0 ? IssueStatus.RESOLVED : issue.status)
+
+            return {
+                issue,
+                count,
+                actualStatus,
+            }
+        }))
+
+        // Filter issues based on the actual calculated status
+        const filteredIssues = issuesWithCounts.filter(({ actualStatus }) => {
+            if (!status || status.length === 0) {
+                return true
+            }
+            
+            return status.includes(actualStatus)
+        })
+
+        const populatedIssues = await Promise.all(filteredIssues.map(async ({ issue, count, actualStatus }) => {
             const flowVersion = await flowVersionService(log).getLatestLockedVersionOrThrow(issue.flowId)
+            if (!isNil(issue.stepName)) {
+                issue.step = await getStepFromFlow(issue.flowId, issue.stepName ?? '', log) 
+            }
+            
             return {
                 ...issue,
                 flowDisplayName: flowVersion.displayName,
+                count,
+                status: actualStatus,
```

Comment:
If you are using actualStatus there, then issue.status=RESOLVED in the if statement block for count = 0 is not used